File: CSharp/Yow.CoD.Finance.Domain/Contracts/Receipt.cs
using System;

namespace Yow.CoD.Finance.Domain.Contracts
{
    public class Receipt
    {
        public Receipt(Guid aggregateId, int version)
        {
            if(aggregateId == Guid.Empty) throw new ArgumentException("AggregateId must be non default value", nameof(aggregateId));

            AggregateId = aggregateId;
            Version = version;
        }
        public Guid AggregateId { get; }
        public int Version { get; }
    }
}

File: java/src/main/java/com/leecampbell/cod/domain/contracts/Receipt.java
package com.leecampbell.cod.domain.contracts;

import java.util.UUID;

public class Receipt {
    private final UUID aggregateId;
    private final int version;

    public Receipt(UUID aggregateId, int version) {
        if (aggregateId == null)
            throw new IllegalArgumentException("AggregateId must be non null");

        this.aggregateId = aggregateId;
        this.version = version;
    }

    public UUID aggregateId() {
        return this.aggregateId;
    }

    public int version() {
        return this.version;
    }
}